### PR TITLE
M.pdf option

### PR DESCRIPTION
**Pull request by mikephoran**
Added PDF option. Deletes all screenshots on phantom startup and shutdown.  Also checks for screenshots older than 1 minutes every minute and deletes them. 

Fixed a number of bad javascript practices in the original repo.

### CODE REVIEW

File: lib/utils.js [hunk 1]
```diff
@@ -19,4 +20,35 @@ exports.md5 = function(str) {
...
-};
\ No newline at end of file
+};
+
+/**
+ * Delete all files in [dirPath] older than [ttl].  Does not modify subdirectories.
+ */
+
+exports.eraseOldFiles = function(dirPath, ttl) {
+  ttl = typeof ttl !== undefined ? ttl : 0;
```
Comment (Woodya) [inline lib/utils.js]: i'd just use `ttl || 0` personally.

File: lib/utils.js [hunk 2]
```diff
@@ -19,4 +20,35 @@ exports.md5 = function(str) {
...
+  var files = fs.readdirSync(dirPath);
```
Comment (Woodya) [inline lib/utils.js]: files var seems extraneous. just fs.readdir().forEach

File: lib/utils.js [hunk 3]
```diff
@@ -19,4 +20,35 @@ exports.md5 = function(str) {
...
+  files.forEach(function(file) {
+    var filePath = path.join(dirPath, file);
+    var stats = fs.statSync(filePath);
+    if (stats.isFile() && file.slice(0,10) === 'screenshot') {
+      var now = new Date().getTime();
```
Comment (Woodya) [inline lib/utils.js]: just calculate the 'expiration time outside the loop as now() - ttl
then check each mtime vs that.

File: routes/index.js [hunk 4]
```diff
@@ -1,80 +1,54 @@
...
-      // asynchronous
-      res.send('Will post screenshot to ' + url + ' when processed');
-      postImageToUrl(filePath, url, callback);
-    } else {
-      // synchronous
-      sendImageInResponse(filePath, res, callback);
+    // add x-headers to options
+    var xheaders = {};
+    for (var key in req.headers) {
+      if (req.headers.hasOwnProperty(key) && key.slice(0,2).toLowerCase().replace('-', '_') === 'x_') {
```
Comment (Woodya) [inline routes/index.js]: more easily written as `/^x[-_]/i.test(key)`

File: scripts/rasterizer.js [hunk 5]
```diff
@@ -64,13 +65,15 @@ service = server.listen(port, function(request, response) {
...
   var page = new WebPage();
+
+  page.customHeaders = JSON.parse(request.headers.xheaders);
   page.onResourceError = function(resourceError) {
-    page.error_reason = resourceError.errorString;
+  page.error_reason = resourceError.errorString;
   };
   var delay = request.headers.delay || 0;
-  try {
+   try {
```
Comment (Woodya) [inline scripts/rasterizer.js]: ?

File: scripts/rasterizer.js [hunk 6]
```diff
@@ -78,10 +81,25 @@ service = server.listen(port, function(request, response) {
...
         value = (value == 'false') ? false : ((value == 'true') ? true : value);
-        page.settings[pageSettings[name]] = value;
+        page.settings[setting] = value;
+      }
+    });
+    if (path.substr(-4) === '.pdf') {
+      //Example values for pdf param: "5in*7.5in", "10cm*20cm", "A4", "Letter"
+      var pdf = request.headers.pdf || defaultPDF;
+      var wxh = pdf.split('*');
+      page.paperSize = wxh.length === 2 ? {
```
Comment (Woodya) [inline scripts/rasterizer.js:124]: is paper-size not modifiable via url params?

File: lib/utils.js (hunk 1 above)
Comment (mikephoran) [inline lib/utils.js]: ha. Weird I almost always do the check your way too, not sure why I overcomplicated it here.  :)

File: lib/utils.js (hunk 3 above)
Comment (mikephoran) [inline lib/utils.js]: good call.

File: lib/utils.js (hunk 2 above)
Comment (mikephoran) [inline lib/utils.js]: done

File: routes/index.js (hunk 4 above)
Comment (mikephoran) [inline routes/index.js]: Ya good point.

File: scripts/rasterizer.js (hunk 5 above)
Comment (mikephoran) [inline scripts/rasterizer.js]: fixed

File: scripts/rasterizer.js (hunk 6 above)
Comment (mikephoran) [inline scripts/rasterizer.js:124]: Don't think I understand this question.   The variable 'pdf' is passed to screenshot via parameter.  This variable determines the paper-size of the PDF.

**Comment by mikephoran** [comment]
@Woodya I squashed all my commits down to one since your last code review.

**Pull request merged**